File: go-gin/cmd/repository.go
package main

import (
	"context"
	"time"

	"github.com/jmoiron/sqlx"
)

type ModelRepository interface {
	Insert(ctx context.Context, model Model) error
	FindByID(ctx context.Context, id string) (*Model, error)
}

type Model struct {
	ID           *string    `json:"id" db:"id"`
	Text         string     `json:"text" db:"text"`
	CreationDate *time.Time `json:"creationDate" db:"creation_date"`
}

type modelRepository struct {
	db *sqlx.DB
}

func NewRepository(db *sqlx.DB) ModelRepository {
	return &modelRepository{db}
}

func (r *modelRepository) Insert(ctx context.Context, model Model) error {
	query := "INSERT INTO model(id, text) VALUES (:id, :text)"
	_, err := r.db.NamedExecContext(ctx, query,
		map[string]interface{}{
			"id":   model.ID,
			"text": model.Text,
		})
	return err
}

func (r *modelRepository) FindByID(ctx context.Context, id string) (*Model, error) {
	query := "SELECT * FROM model WHERE id = $1"
	model := Model{}
	if err := r.db.GetContext(ctx, &model, query, id); err != nil {
		return nil, err
	}
	return &model, nil
}

File: node-express/src/Repository.ts
import {Model} from "./Model"
import {Pool} from "pg"
import {SQL} from "sql-template-strings"

const pool = new Pool({
  host: 'localhost',
  database: 'test',
  user: 'root',
  password: 'root',
  port: 5432,
  max: 10,
  min: 10,
  idleTimeoutMillis: 1800_000 // 30m
});

export class Repository {
  insert = async (model: Model) => {
    try {
      const res = await pool.query(
        SQL`INSERT INTO model(id, text) VALUES (${model.id}, ${model.text})`
      );

      for (let row of res.rows) {
        console.log(row);
      }
    } catch (err) {
      console.error(err);
    }
  }

  findById = async (id: String) => {
    try {
      const res = await pool.query(SQL`SELECT * FROM model WHERE id = ${id}`);

      for (let row of res.rows) {
        console.log(row);
      }
    } catch (err) {
      console.error(err);
    }
  }
}
